charts/kserve-resources/templates/ingress_gateway.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if .Values.kserve.controller.gateway.ingressGateway.enableGatewayApi }}
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: kserve-ingress-gateway
+  namespace: {{ .Release.Namespace }}
+spec:
+  gatewayClassName: envoy
+  listeners:
+   - name: http
+     port: 80
+     protocol: HTTP
+     allowedRoutes:
+       namespaces:
+         from: All
```

Comment:
Not sure if kserve should ship this gateway by default, user might already have gateway defined in their setup. Let's add a flag to create the kserve ingress gateway resource.

Reply:
Done